gateway-server/src/main/java/org/apache/knox/gateway/GatewayServer.java

```diff
@@ -933,6 +936,20 @@ private synchronized void internalActivateArchive( Topology topology, File warDi
     }
   }
 
+  private void processApplicationPathAliases(File warDir, Topology topology) {
+    final Map<String, Collection<String>> applicationPathAliases = config.getApplicationPathAliases();
+    if (!applicationPathAliases.isEmpty()) {
```

Comment:
Why do we need the `isEmpty` before iterating over the map?

Reply:
We don't need it, just defensive programming. I'd leave it for now, it does not hurt or cause any issues.

Reply:
Then at least let's make it a null check instead of `isEmpty`. That would protect us against NPEs, while this doesn't defend against anything. 
Personally if `config.getApplicationPathAliases()` is guaranteed to be not null, I wouldn't even use a null check, but if I had to decide between those 2 then I would choose a null check.